### PR TITLE
Format flag issue

### LINKED ISSUE #58: Format flag causes exit 1
Same issues as #52. I solved it for --format text and gnu but it still happens for JSON and XML. 

Both JSON and XML come back with an error field that says it found no errors. Fix and test in progress.

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
- Fixes #58 
- Fixed issue with ```--format``` flag return error when none occurred
- Updated Changelog
- Added Gitlab CI integration

### CODE REVIEW

File: html5validator/validator.py [hunk 1]
```diff
@@ -150,8 +152,26 @@ def validate(self, files):
...
-            LOGGER.error(stderr)
+        if e:
+            LOGGER.debug(e)
+            if self.format == "text":
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "gnu":
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "json" and e[0] != u'{"messages":[]}':
```
Comment (svenkreiss) [inline html5validator/validator.py]: Can this be absorbed into an ignore pattern?

File: html5validator/validator.py [hunk 2]
```diff
@@ -150,8 +152,26 @@ def validate(self, files):
...
+                LOGGER.error(e)
+                return len(e)
+            elif self.format == "xml" and len(e) > 4:
```
Comment (svenkreiss) [inline html5validator/validator.py]: Same here?

And then is it necessary to treat the different formatting options differently? I would prefer they all get handled the same way.

File: html5validator/validator.py (hunk 1 above)
Comment (Cyb3r-Jak3) [reply html5validator/validator.py]: The JSON message has been successfully added into the ignore pattern

File: html5validator/validator.py (hunk 2 above)
Comment (Cyb3r-Jak3) [reply html5validator/validator.py]: I agree that treating the different format options is bad idea, I only did it because it was a primitive way to fix the formatting issues. 
I am currently working on the XML formatting but running into errors where not all parts are being removed.

Comment (Cyb3r-Jak3) [reply html5validator/validator.py]: The latest commit treats all format types the same.

File: html5validator/validator.py [hunk 3]
```diff
@@ -144,14 +157,24 @@ def validate(self, files):
 
         e = stderr.splitlines()
 
+        # Removes any empty items in the list
+        e = list(filter(None, e))
+
+        # Prevents removal of xml tags if there are errors
+        if self.format == "xml" and len(e) > 4:
+            self.ignore = self.ignore[3:]
```
Comment (Cyb3r-Jak3) [inline html5validator/validator.py]: This is needed because otherwise 3 tags are cut from xml output.

File: html5validator/validator.py [hunk 4]
```diff
@@ -144,14 +157,24 @@ def validate(self, files):
...
             e = [l for l in e if i not in l]
         for i in self.ignore_re:
             regex = re.compile(i)
             e = [l for l in e if not regex.search(l)]
 
-        if stderr:
-            LOGGER.error(stderr)
+        if e:
+            LOGGER.error(e)
+            return len(e)
```
Comment (Cyb3r-Jak3) [inline html5validator/validator.py]: All format types are now treated the same for output

File: html5validator/validator.py [hunk 5]
```diff
@@ -13,7 +13,17 @@
...
+    r'\APicked up _JAVA_OPTIONS:.*',
+    r'\ADocument checking completed. No errors found.*',
+]
+
+DEFAULT_IGNORE = [
+    '</messages>',
+    '<?xml version=\'1.0\' encoding=\'utf-8\'?>',
+    '<messages xmlns="http://n.validator.nu/messages/">',
+    '{"messages":[]}',
+]
```
Comment (Cyb3r-Jak3) [inline html5validator/validator.py]: Added these to cut out items that out be in the format.

File: html5validator/tests/test_simple.py [hunk 6]
```diff
@@ -105,10 +105,43 @@ def test_stack_size():
...
 if __name__ == '__main__':
     test_valid()
     test_invalid()
     test_return_value()
     test_angularjs()
     test_multiple_ignoreres()
     test_ignore_and_ignorere()
+    test_stack_size()
+    test_valid_format_flags()
+    test_invalid_format_flags()
```
Comment (Cyb3r-Jak3) [inline html5validator/tests/test_simple.py]: Added tests to make sure the format flags work

File: html5validator/validator.py [hunk 7]
```diff
@@ -144,14 +157,24 @@ def validate(self, files):
...
 
-        if stderr:
-            LOGGER.error(stderr)
+        if e:
+            LOGGER.error(e)
+            return len(e)
         else:
             LOGGER.info('All good.')
-        return len(e)
+            return len(e)
```
Comment (svenkreiss) [inline html5validator/validator.py]: Was this change intentional?

Comment (Cyb3r-Jak3) [reply html5validator/validator.py]: It was but I fixed it now.
Before both the if and else had their own return but I moved it back so there is only one.

File: html5validator/validator.py (hunk 3 above)
Comment (Cyb3r-Jak3) [reply html5validator/validator.py]: Done
